aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -69,20 +69,16 @@
     private final Merger merger;
     private final DataStoreConfigurationProvider dataStoreConfigurationProvider;
     private final String[] modelNames;
-
-    private SyncProcessor(
-            ModelProvider modelProvider,
-            ModelSchemaRegistry modelSchemaRegistry,
-            SyncTimeRegistry syncTimeRegistry,
-            AppSync appSync,
-            Merger merger,
-            DataStoreConfigurationProvider dataStoreConfigurationProvider) {
-        this.modelProvider = Objects.requireNonNull(modelProvider);
-        this.modelSchemaRegistry = Objects.requireNonNull(modelSchemaRegistry);
-        this.syncTimeRegistry = Objects.requireNonNull(syncTimeRegistry);
-        this.appSync = Objects.requireNonNull(appSync);
-        this.merger = Objects.requireNonNull(merger);
-        this.dataStoreConfigurationProvider = dataStoreConfigurationProvider;
+    private final QueryPredicateProvider queryPredicateProvider;
+
+    private SyncProcessor(Builder builder) {
```

Comment:
Not that this matters or is worth changing but is there a reason you switched from passing in params to passing in the Builder instance? We do the opposite elsewhere in the code.

Reply:
I've seen both patterns in our code base.  I like the single `Builder` parameter approach a little better, because then we don't end up with a constructor that takes a ton of parameters, and one of the main reasons for using the Builder pattern is to avoid constructors with lots of parameters :).   I'm open to other thoughts on why the other pattern might be better though.

Reply:
My argument for all-args constructor is that the constructor should not know any detail about the builder, or how it is being constructed. However, if the constructor is going to be `private` anyway, then it really doesn't matter. @richardmcclellan's point about not having to maintain the constructor argument list makes sense to me.